include/rocksdb/compaction_policy.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <memory>
+
+#include "rocksdb/listener.h"
+
+namespace rocksdb {
+
+struct CompactionInfo {
+  int output_level;
+};
+
+struct CompactionOutputInfo {
+  Slice next_key;
```

Comment:
Does this include range deletions? For example imagine we're outputting (1) key "a", (2) range tombstone "[x, y)", (3) key "z". If we only consider point keys then we might accept "a" and reject "z". Then I guess we are stuck with "a" and "[x, y)" together in the first output file, and we'll have to assign it the very wide range "[a, y]".

This is probably unrelated to the intent of your diff. It's just something on the `DeleteRange` TODO list for a long time and I wonder if this interface might help.

Reply:
@ajkr thanks! I'm not familiar with how compaction deal with delete ranges, but I'll check and get back to you.